DAL/EmployeeDAL.cs

```
@@ -0,0 +1,218 @@
+                                         updatedEmployee.MobileNumber;
+        employeeToUpdate.JoiningDate = updatedEmployee.JoiningDate == DateTime.MinValue ? null :
+                                        updatedEmployee.JoiningDate ?? updatedEmployee.JoiningDate;
+        employeeToUpdate.LocationId = updatedEmployee.LocationId == -1 ? null :
+                                   updatedEmployee.LocationId ?? employeeToUpdate.LocationId;
+        employeeToUpdate.JobTitle = updatedEmployee.JobTitle == "--d" ? null :
+                                    string.IsNullOrWhiteSpace(updatedEmployee.JobTitle) ? employeeToUpdate.JobTitle :
+                                    updatedEmployee.JobTitle;
+        employeeToUpdate.DepartmentId = updatedEmployee.DepartmentId == -1 ? null :
+                                     updatedEmployee.DepartmentId ?? employeeToUpdate.DepartmentId;
+        employeeToUpdate.AssignManager = updatedEmployee.AssignManager == "--d" ? null :
+                                          string.IsNullOrWhiteSpace(updatedEmployee.AssignManager) ? employeeToUpdate.AssignManager :
+                                          updatedEmployee.AssignManager;
+        employeeToUpdate.AssignProject = updatedEmployee.AssignProject == "--d" ? null :
+                                          string.IsNullOrWhiteSpace(updatedEmployee.AssignProject) ? employeeToUpdate.AssignProject :
+                                          updatedEmployee.AssignProject;
+
+        int index = existingEmployees.FindIndex(emp => emp.EmpNo == empNo);
+        existingEmployees[index] = employeeToUpdate;
+        jsonUtils.WriteJSON(existingEmployees, filePath);
+        return true;
+    }
+
+    public bool Delete(string empNo)
+    {
+        List<Employee> existingEmployees = jsonUtils.ReadJSON<Employee>(filePath);
+        Employee employeeToDelete = existingEmployees.FirstOrDefault(employee => employee.EmpNo == empNo);
+        if (employeeToDelete != null)
```

you can throw exception if record not found